testRouter.js

```diff
@@ -44,149 +40,167 @@ const transformDestination = dest => {
               }
-            } else {
+            } catch (err) {
+              errorFound = true;
               response.dest_transformed_payload = {
-                error:
-                  "error encountered in user_transformation stage. Aborting."
+                error: err.message || JSON.stringify(err)
               };
             }
+          } else {
+            response.dest_transformed_payload = {
+              error: "error encountered in user_transformation stage. Aborting."
+            };
           }
-          // const transformerStatuses = [];
-          if (stage.dest_transform && stage.send_to_destination) {
-            // send event to destination only after transformation
-            if (!errorFound) {
-              const destResponses = [];
-              const destResponseStatuses = [];
-
-              const transformedPayloads = response.dest_transformed_payload;
-              // eslint-disable-next-line no-restricted-syntax
-              for (const payload of transformedPayloads) {
-                // eslint-disable-next-line no-await-in-loop
-                const parsedResponse = await sendToDestination(dest, payload);
-
-                destResponses.push(parsedResponse.response);
-                destResponseStatuses.push(parsedResponse.status);
-
-                // TODO: Use updated handleResponseTransform function
-                // Removing the below part, because transformerStatus is not
-                // currently being returned by test api response
-
-                // call response transform here
-                // const ctxMock = {
-                //   request: {
-                //     body: parsedResponse
-                //   }
-                // };
-                // handleResponseTransform(version, dest, ctxMock);
-                // const { output } = ctxMock.body;
-                // transformerStatuses.push(output.status);
+        }
+        // const transformerStatuses = [];
+        if (stage.dest_transform && stage.send_to_destination) {
+          // send event to destination only after transformation
+          if (!errorFound) {
+            const destResponses = [];
+            const destResponseStatuses = [];
+
+            const transformedPayloads = response.dest_transformed_payload;
+            // eslint-disable-next-line no-restricted-syntax
+            for (const payload of transformedPayloads) {
+              // eslint-disable-next-line no-await-in-loop
+              const parsedResponse = await sendToDestination(dest, payload);
+
+              let contentType = "";
+              let response = "";
+              if (parsedResponse.headers) {
+                contentType = parsedResponse.headers["content-type"];
+                if (
+                  contentType.includes("text") ||
+                  contentType.toLowerCase().includes("application/xml") ||
+                  contentType.toLowerCase().includes("application/json")
```

Comment:
I would create a separate list of supported content types.

Reply:
makes sense. will put them in a list